packages/server/src/course/course.controller.ts

```diff
@@ -26,9 +27,46 @@ export class CourseController {
 
   @Get(':id')
   async get(@Param('id') id: number): Promise<GetCourseResponse> {
-    return await CourseModel.findOne(id, {
-      relations: ['officeHours', 'queues', 'queues.staffList'],
+    // TODO: for all course endpoint, check if they're a student or a TA
+    const course = await CourseModel.findOne(id, {
+      relations: ['officeHours'],
     });
+
+    const now = new Date();
+    const MS_IN_MINUTE = 60000;
+
+    const rooms = course.officeHours
+      .filter(
+        (e) =>
+          e.startTime.valueOf() - 15 * MS_IN_MINUTE < now.valueOf() &&
+          e.endTime.valueOf() + 1 * MS_IN_MINUTE > now.valueOf(),
+      )
+      .map((e) => e.room);
+
+    const queues = await QueueModel.find({
+      where: {
+        courseId: id,
+      },
+      relations: ['staffList'],
+    });
+
+    const nonEmptyQueues = queues.filter((e) => e.staffList.length > 0);
```

Comment:
I think this is the best way of doing things for now but I'm still not sure how we'll be able to return the upcoming office hours from #76 if the upcoming hours don't have any staff in them yet?
Maybe we'll have another endpoint to return the upcoming office hours

Reply:
the upcoming queues are returned cause of the funky math tho